### DISCUSSION TITLE
How can I a read in a DEF from a previous run and then route it?

### DISCUSSION THREAD

**Discussion by mumu157** [The-OpenROAD-Project/OpenROAD-flow-scripts#1165] [Q&A]
1、I used OpenROAD-flow-scripts to achieve synthesis and floor planning.
2、Then, I dumped DEF and LEF files and used my own placement tool to obtain the .def file.
3、Now, I'm wondering how I can proceed to the next steps by flow, such as clock tree synthesis (CTS), routing, and more.  

I have seen a similar solution in OpenLane, which can be found at https://github.com/The-OpenROAD-Project/OpenLane/discussions/1256.

Thank you.

**Comment by maliberty** [comment]
The flow doesn't automatically support that but you could modify the script for the next step in the flow to use read_lef/read_def, instead of read_odb, to load your files.
